internal/server/server.go

```diff
@@ -68,6 +68,43 @@ func FromCluster(in *redskyv1alpha1.Experiment) (redskyapi.ExperimentName, *reds
 		})
 	}
 
+	out.Constraints = nil
```

Comment:
nitpick: Initialize the capacity of `out.Constraints`:

```go
out.Constraints = make([]redskyapi.Constraint, 0, len(in.Spec.Constraints))
```

You can do the same thing on line 108 for metrics as well.

Reply:
I had considered that (in several places), but ultimately the numbers are so small (e.g. this function is called once over the lifetime of an experiment on slices that contain only a few elements) that I opted for the simple code. In this particular case the `nil` slice also serializes differently then an empty slice (which impacts test and network representations).